---
title: "RNA_mT_Hv3_stats"
output:
  html_notebook: default
  pdf_document: default
---

# Load necessary libraries
```{r}
library(dplyr)
library(ggplot2)
library(patchwork)
library(purrr)
library(stringr)
```

# Basic RNA mT stats
## Read data from tsv file
```{r}
# Data contains the header
read.table(file = '~/Desktop/mT/new-masterTable/Target_slicing/formatted_T_RNA_Hv3_ICmT.tsv',
           sep = '\t', header = TRUE) %>% 
  as_tibble() -> df_rna_basic
```


### Barplot
```{r}
#pdf(file = "IC_T_RNA_mT_sample_analysis_pre_and_post.pdf", width = 6, height = 4)
df_rna_basic %>%
  select(transcript_id, sampleN) %>% 
  distinct(transcript_id, sampleN) %>% 
  group_by(sampleN) %>% 
  summarise(total_count=n_distinct(transcript_id)) %>%
  filter(sampleN <= 10) %>% 
  mutate(sampleN = as.character(sampleN)) %>% 
  add_row(sampleN=paste0(11, "-", (df_rna_basic %>% 
                                     select(sampleN) %>%
                                     max())),
          total_count=(df_rna_basic %>%
                         select(transcript_id, sampleN) %>%
                         distinct(transcript_id, sampleN) %>%
                         group_by(sampleN) %>%
                         summarise(total_count=n_distinct(transcript_id)) %>%
                         filter(sampleN > 10) %>% 
                         select(total_count) %>% 
                         sum())) %>% 
  mutate(percentage = prop.table(total_count)) %>%
  ggplot(., aes(x=factor(sampleN, levels=c("1", "2", "3", "4", "5", "6", "7",
                                           "8", "9", "10", "11-90")),
                y=percentage)) + 
  geom_bar(stat="identity", fill="steelblue") +
  geom_text(aes(label=scales::comma(total_count)), vjust=-0.3, size=3.5) +
  theme_minimal() +
  #scale_x_continuous(breaks = seq(1, 20, by = 1)) +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1)) +
  ylab('Percentage') +
  xlab('sampleN') +
  theme(legend.position = "none",
        axis.ticks = element_line(colour = "black"),
        plot.title = element_text(hjust = 0.5, size=11),
        plot.subtitle = element_text(hjust = 0.5, size=9, face="italic"),
        panel.background = element_rect(fill = NA),
        panel.grid.major.x = element_line(colour = NA),
        panel.grid.minor.x = element_line(colour = NA),
        axis.text = element_text(colour="black",
                                 face="bold")) +
  ggtitle(label = "Barchart representing number of lnc-RNA intron chain transcripts\npresent in different numbers of samples (sampleN)")
```

# mT analysis with sample data
## Read data from tsv file
```{r}
# Data DO NOT contain header
read.table(file = '~/Desktop/mT/new-masterTable/Target_slicing/T_RNA_ICmT_Hv3_sample_stats.tsv', sep = '\t', header = FALSE) %>% 
  as_tibble() -> df_rna_samples

colnames(df_rna_samples) <- c('transcript_id',
                              'tissue',
                              'stage',
                              'technology',
                              'capture_status',
                              'biological_replicate',
                              'technical_replicate')
```

## Determine number of transcripts for ea platform (multi - both ont and pacBio)
```{r}
multi_rna_transcripts <- df_rna_samples %>%
  select(transcript_id, technology) %>%
  filter(technology=="ont") %>%
  select(transcript_id) %>%
  intersect(., (df_rna_samples %>%
                  select(transcript_id, technology) %>%
                  filter(technology=="pacBioSII") %>%
                  select(transcript_id)))

technology_rna_df <- data.frame(technology = c("multi", "ont", "pacBioSII"),
                            number_of_transcripts = c(multi_rna_transcripts %>%
                                                        nrow(),
                                                      df_rna_samples %>%
                                                        select(transcript_id, technology) %>%
                                                        filter(technology=="ont") %>%
                                                        select(transcript_id) %>%
                                                        setdiff(., multi_rna_transcripts) %>%
                                                        nrow(),
                                                      df_rna_samples %>%
                                                        select(transcript_id, technology) %>%
                                                        filter(technology=="pacBioSII") %>%
                                                        select(transcript_id) %>%
                                                        setdiff(., multi_rna_transcripts) %>%
                                                        nrow()))

# Plot the results
technology_rna_df %>%
  mutate(percent = prop.table(number_of_transcripts)) %>% 

  ggplot(data=.,
         aes(x=reorder(technology, -percent),
             y=percent,
             fill=technology)) + 
  geom_bar(stat="identity", position = "dodge") +
  geom_text(aes(label = scales::comma(number_of_transcripts)),
            position = position_dodge(width = 1),
            vjust=-0.5,
            size = 3) +
  theme_minimal() +
  ggtitle(label="Number of lnc-RNA transcripts aquired using ont, pacBio or both technologies") +
  scale_fill_brewer(palette="Set2") +
  ylab('Percentage') +
  xlab('Technology') +
  theme(legend.position = "none") +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1)) -> p_rna_technology

#pdf(file = "IC_T_RNA_transcripts_by_technology.pdf", width = 6, height = 4)
p_rna_technology



##### Stacked barplot
# Plot the results
technology_rna_df %>%
  mutate(percent = prop.table(number_of_transcripts)) %>% 

  ggplot(data=.,
         aes(x=1,#reorder(technology, -percent),
             y=percent,
             fill=factor(technology, levels=c("ont", "multi", "pacBioSII")))) + 
  geom_bar(stat="identity") +
   geom_text(aes(label = paste0(technology, " (", scales::comma(number_of_transcripts), ")")),
             position=position_stack(vjust=0.5),
             size = 4) +
  theme_minimal() +
  scale_fill_manual(values=c("#377EB8", "#4DAF4A", "#FFFF33")) +
  ylab('Percentage') +
  xlab('Technology') +
  theme(legend.position = "none",
        axis.ticks.y = element_line(colour = "black"),
        plot.title = element_text(hjust = 0.5,
                                  size=9.0),
        panel.background = element_rect(fill = NA),
        panel.grid.major.x = element_line(colour = NA),
        panel.grid.minor.x = element_line(colour = NA),
        axis.text.x = element_blank(),
        axis.text.y = element_text(colour="black",
                                  face="bold")) +
  ggtitle(label="Number of lnc-RNA transcripts aquired\nusing ont, pacBio or both technologies") +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1)) -> p_rna_technology2

#pdf(file = "IC_T_RNA_transcripts_by_technology_stacked.pdf", width = 3, height = 4)
p_rna_technology2
```

## Tissues - Correlation matrix
### Pre-capture
```{r}
pre_cap_rna_matrix_df <- df_rna_samples %>%
  filter(capture_status=="pre-capture") %>%    # Only pre-capture
  distinct(transcript_id, tissue, stage) %>%   # Leave only unique tissue per transcript > don't include technology
  mutate(tissue=paste0(tissue," (", stage, ")")) %>% 
  select(transcript_id, tissue) %>% 
  group_by(tissue) %>% 
  summarize(transcripts = list(transcript_id)) %>% 
  cross_join(x=., y=., suffix = c("1", "2")) %>% 
  rowwise() %>% 
  mutate(intersecting = list(intersect(x=transcripts1, y=transcripts2))) %>% 
  mutate(intersecting_no = length(intersecting)) %>% 
  mutate(percentage = intersecting_no/length(transcripts1) * 100) %>% 
  mutate(transcripts1_number = length(transcripts1))


pre_cap_rna_matrix_df %>% 
  ggplot(data=.) +
  geom_tile(data = pre_cap_rna_matrix_df %>% filter(percentage == 100),
            aes(x=tissue1,
                y=tissue2), fill="#3591AC") +
  geom_tile(data = pre_cap_rna_matrix_df %>% filter(percentage < 100),
            aes(x=tissue1,
                y=tissue2,
                fill = percentage)) +
  geom_text(size=5,
            aes(x=tissue1,
                y=tissue2,
                label = (scales::percent(percentage/100, accuracy = 0.1L)))) +
  geom_text(size=3.2,
            aes(x=tissue1,
                y=tissue2,
                label = paste0("\n\n",
                               "(", scales::comma(intersecting_no), "/", scales::comma(transcripts1_number), ")"))) +
  scale_fill_gradient(low = "white",
                      high = "#3EA8C8") +
   theme(axis.title.x = element_blank(),
         axis.title.y = element_blank(),
         axis.ticks = element_line(colour = NA),
         plot.title = element_text(hjust = 0.5),
         panel.background = element_rect(fill = NA),
         panel.grid.major = element_line(colour = NA),
         axis.text.x = element_text(colour="black",
                                    face="bold"),
         axis.text.y = element_text(colour="black",
                                    face="bold")) +
  labs(title = "Lnc-RNA transcripts shared between tissues (pre-capture)", fill = "% of transcripts") -> p_rna_pre_cap_heatmap


#pdf(file="IC_T_RNA_pre_cap_heatmap.pdf", width = 16, height = 8)
p_rna_pre_cap_heatmap
```

### Post-capture
```{r}
post_cap_rna_matrix_df <- df_rna_samples %>%
  filter(capture_status=="post-capture") %>%    # Only post-capture
  distinct(transcript_id, tissue, stage) %>%   # Leave only unique tissue per transcript > don't include technology
  mutate(tissue=paste0(tissue," (", stage, ")")) %>% 
  select(transcript_id, tissue) %>% 
  group_by(tissue) %>% 
  summarize(transcripts = list(transcript_id)) %>% 
  cross_join(x=., y=., suffix = c("1", "2")) %>% 
  rowwise() %>% 
  mutate(intersecting = list(intersect(x=transcripts1, y=transcripts2))) %>% 
  mutate(intersecting_no = length(intersecting)) %>% 
  mutate(percentage = intersecting_no/length(transcripts1) * 100) %>% 
  mutate(transcripts1_number = length(transcripts1))

post_cap_rna_matrix_df %>% 
  ggplot(data=.) +
  geom_tile(data = post_cap_rna_matrix_df %>% filter(percentage == 100),
            aes(x=tissue1,
                y=tissue2), fill="#3FA9AF") +
  geom_tile(data = post_cap_rna_matrix_df %>% filter(percentage < 100),
            aes(x=tissue1,
                y=tissue2,
                fill = percentage)) +
  geom_text(size=5,
            aes(x=tissue1,
                y=tissue2,
                label = (scales::percent(percentage/100, accuracy = 0.1L)))) +
  geom_text(size=3.2,
            aes(x=tissue1,
                y=tissue2,
                label = paste0("\n\n",
                               "(", scales::comma(intersecting_no), "/", scales::comma(transcripts1_number), ")"))) +
  scale_fill_gradient(low = "white",
                      high = "#49C4CB") +
   theme(axis.title.x = element_blank(),
         axis.title.y = element_blank(),
         axis.ticks = element_line(colour = NA),
         plot.title = element_text(hjust = 0.5),
         panel.background = element_rect(fill = NA),
         panel.grid.major = element_line(colour = NA),
         axis.text.x = element_text(colour="black",
                                    face="bold"),
         axis.text.y = element_text(colour="black",
                                    face="bold")) +
  labs(title = "Lnc-RNA transcripts shared between tissues (post-capture)", fill = "% of transcripts") -> p_rna_post_cap_heatmap


#pdf(file="IC_T_RNA_post_cap_heatmap.pdf", width = 16, height = 8)
p_rna_post_cap_heatmap

```

### Pre&Post-capture
```{r}
preANDpost_cap_rna_matrix_df <- df_rna_samples %>%
  distinct(transcript_id, tissue, stage) %>%   # Leave only unique tissue per transcript > don't include technology
  mutate(tissue=paste0(tissue," (", stage, ")")) %>% 
  select(transcript_id, tissue) %>% 
  group_by(tissue) %>% 
  summarize(transcripts = list(transcript_id)) %>% 
  cross_join(x=., y=., suffix = c("1", "2")) %>% 
  rowwise() %>% 
  mutate(intersecting = list(intersect(x=transcripts1, y=transcripts2))) %>% 
  mutate(intersecting_no = length(intersecting)) %>% 
  mutate(percentage = intersecting_no/length(transcripts1) * 100) %>% 
  mutate(transcripts1_number = length(transcripts1))

preANDpost_cap_rna_matrix_df %>% 
  ggplot(data=.) +
  geom_tile(data = preANDpost_cap_rna_matrix_df %>% filter(percentage == 100),
            aes(x=tissue1,
                y=tissue2), fill="#0470B4") +
  geom_tile(data = preANDpost_cap_rna_matrix_df %>% filter(percentage < 100),
            aes(x=tissue1,
                y=tissue2,
                fill = percentage)) +
  geom_text(size=5,
            aes(x=tissue1,
                y=tissue2,
                label = (scales::percent(percentage/100, accuracy = 0.1L)))) +
  geom_text(size=3.2,
            aes(x=tissue1,
                y=tissue2,
                label = paste0("\n\n",
                               "(", scales::comma(intersecting_no), "/", scales::comma(transcripts1_number), ")"))) +
  scale_fill_gradient(low = "white",
                      high = "#0483D2") +
   theme(axis.title.x = element_blank(),
         axis.title.y = element_blank(),
         axis.ticks = element_line(colour = NA),
         plot.title = element_text(hjust = 0.5),
         panel.background = element_rect(fill = NA),
         panel.grid.major = element_line(colour = NA),
         axis.text.x = element_text(colour="black",
                                    face="bold"),
         axis.text.y = element_text(colour="black",
                                    face="bold")) +
  labs(title = "Lnc-RNA transcripts shared between tissues (pre&post-capture)", fill = "% of transcripts") -> p_rna_preANDpost_cap_heatmap


#pdf(file="IC_T_RNA_preANDpost_cap_heatmap.pdf", width = 16, height = 8)
p_rna_preANDpost_cap_heatmap
```


# ALL vs RNA IC mT transcripts
## Read data
```{r}
# ALL IC mT transcripts
read.table(file = '~/Desktop/mT/new-masterTable/formatted_Hv3_ICmT.tsv',
           sep = '\t', header = TRUE) %>% 
  as_tibble() -> df_all_ICmT

# RNA IC mT transcripts (from putative and lncRNA targets)
read.table(file = '~/Desktop/mT/new-masterTable/Target_slicing/formatted_T_RNA_Hv3_ICmT.tsv',
           sep = '\t', header = TRUE) %>% 
  as_tibble() -> df_rna_ICmT
```

## Comparison plot
```{r}
df_all_ICmT %>% 
  distinct(transcript_id) %>% 
  nrow() -> ALL_transcripts

df_rna_ICmT %>% 
  distinct(transcript_id) %>% 
  nrow() -> RNA_transcripts

tibble(x = c("masterTable", "RNA masterTable"), y = c(ALL_transcripts, RNA_transcripts)) %>% 
  mutate(percent = c(1, RNA_transcripts/ALL_transcripts)) %>%
  ggplot(aes(x = x, y = percent, fill=x)) +
    geom_bar(stat="identity", position = "dodge") +
  geom_text(aes(label = scales::comma(y)),
            position = position_dodge(width = 0.9),
            vjust=-0.5,
            size = 4) +
  theme_minimal() +
  ggtitle("Number of transcripts in mT and it's RNA slice") +
  scale_fill_manual(values=c("#C76A24", "#0483D2")) +
  ylab('Percentage') +
  theme(axis.title.x=element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.text = element_text(colour = "black", size = 11),
        axis.ticks = element_line(colour = "black"),
        panel.background = element_rect(fill = "white", colour = "grey50"),
        legend.position = "none",
        panel.grid = element_blank()
        ) +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1.05),
                     breaks = c(0, 0.25, 0.5, 0.75, 1)) -> ALL_vs_RNA_plot

#pdf(file = "IC_ALL_vs_T_RNA_plot.pdf", width = 6, height = 5)

ALL_vs_RNA_plot
```


# Targeted/detected chart - ale z naciskiem na % i podziałem na tkanki

## Load data
```{r}
# Read targets from targetDesign file
# Data contains the header
read.table(file = '~/Desktop/mT/new-masterTable/Hv3_CLS3_targetDesign.tsv', sep = '\t', header = TRUE) %>% 
  as_tibble() -> df_targets
df_targets


# mT RNA slice using only lncRNA targets
# Read targets from masterTable with transcript_ids, target_ids and sample tissue
# Data contains the header
read.table(file = '~/Desktop/mT/new-masterTable/Target_slicing/targets_TL_RNA_Hv3_ICmT_with_tissues.tsv', sep = '\t', header = TRUE) %>% 
  as_tibble() -> df_TL_RNA_ICmT_with_tissues
df_TL_RNA_ICmT_with_tissues


# mT RNA slice using lncRNA + putative targets
# Read targets from masterTable with transcript_ids, target_ids and sample tissue
# Data contains the header
read.table(file = '~/Desktop/mT/new-masterTable/Target_slicing/targets_T_RNA_Hv3_ICmT_with_tissues.tsv', sep = '\t', header = TRUE) %>% 
  as_tibble() -> df_T_RNA_ICmT_with_tissues
df_T_RNA_ICmT_with_tissues

```

## Plot for lncRNA alone
```{r}
#pdf(file = "N_TL_RNA_IC_detected_targets_percent2_tkanka.pdf", width = 8, height = 4)
df_targets %>% 
  filter(target_category=="longNonCoding") %>% # Nie ma znaczenia czy pozniej wezmę TL czy samo T, bo w tej linii odfiltrowuję tylko targety lncRNA
  select(target_id, target_set) %>% 
  inner_join(x=.,
            y=(df_TL_RNA_ICmT_with_tissues %>%
                 filter(tissue=="Brain") %>% # Testis, CpoolA, Brain, Placenta, WBlood, Heart, TpoolA, Liver, iPSC
                 filter(technology=="ont") %>% # ont, pacBioSII
                 filter(stage=="Adult") %>% # Adult, Placenta, Embryo
                 filter(capture_status=="post-capture") %>%
                 filter(bioReplicate==3) %>%
                 filter(techReplicate==1) %>%
                 #filter(grepl('chr', chr)) %>%
                 distinct(target) %>%  
                 mutate(status="detected")),
            by=join_by(target_id==target)) %>%
  summarise(count=n(), .by=c(target_set, status)) %>% 
  rbind(df_targets %>% 
          filter(target_category=="longNonCoding") %>% 
          select(target_id, target_set) %>%
          summarise(count=n(), .by=c(target_set)) %>%
          mutate(status="targeted")) %>%
  group_by(target_set) %>% 
  mutate(percentage=count/max(count)) %>% 
  ggplot(data=.,aes(x=reorder(target_set, -percentage),
                    y=percentage,
                    fill=reorder(status, -count))) +
  geom_bar(stat="identity", position=position_dodge()) +
  geom_text(aes(label = paste0(scales::percent(percentage), "\n(", scales::comma(count), ")")),
          position = position_dodge(width=0.9),
          vjust=-0.25,
          size = 2.9) +
  theme_minimal() +
  ylab('Percentage') +
  ggtitle(label = "Targeted to detected ratio for each catalogue (WBlood)") + # Testis, CpoolA, Brain, Placenta, WBlood, Heart, TpoolA, Liver, iPSC
  scale_fill_brewer(palette="Blues", direction = -1) +
  theme(axis.title.x=element_blank(),
        axis.text = element_text(colour = "black", size = 11),
        axis.ticks = element_line(colour = "black"),
        panel.background = element_rect(fill = "white", colour = "grey50"),
        legend.title = element_blank(),
        legend.position = "top",  #c(0.9, 0.9),
        panel.grid = element_blank(),
        plot.title = element_text(hjust = 0.5, size=11)
        ) +
  scale_y_continuous(breaks = c(0, 0.25, 0.50, 0.75, 1.00), limits = c(0, 1.11), labels=scales::percent)
```















# JUNK Load data:
```{r}
df <- read.delim("/home/tomasz/Desktop/mT/RNA_mT_Hv3_sample_stats.tsv",
                 sep="\t",
                 header = FALSE)
#View(df)

colnames(df) <- c('tissue',
                  'stage',
                  'technology',
                  'capture_status',
                  'biological_replicate',
                  'technical_replicate')
```

## Create some plots:
```{r}
df %>%
  group_by(technology) %>%
  summarise(total_count=n()) %>% 
  mutate(percent = prop.table(total_count)) %>% 

  ggplot(data=.,
         aes(x=technology,
             y=percent,
             fill=technology)) + 
  geom_bar(stat="identity", position = "dodge") +
  geom_text(aes(label = scales::comma(total_count)), #label = paste0(total_count)),
            position = position_dodge(width = 1),
            vjust=-0.5,
            size = 3) +
  theme_minimal() +
  scale_fill_brewer(palette="Pastel1") +
  ylab('Percentage') +
  xlab('Technology') +
  theme(legend.position = "none") +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1)) -> p1

df %>%
  group_by(tissue) %>%
  summarise(total_count=n()) %>% 
  mutate(percent = prop.table(total_count)) %>% 

  ggplot(data=.,
         aes(x=reorder(tissue, -percent),
             y=percent,
             fill=tissue)) + 
  geom_bar(stat="identity", position = "dodge") +
  geom_text(aes(label = scales::comma(total_count)),
            position = position_dodge(width = 1),
            vjust=-0.5,
            size = 2) +
  theme_minimal() +
  scale_fill_brewer(palette="Spectral") +
  xlab('Tissue') +
  ylab('Percentage') +
  theme(legend.position = "none") +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,0.3)) -> p2

df %>%
  group_by(stage) %>%
  summarise(total_count=n()) %>% 
  mutate(percent = prop.table(total_count)) %>% 

  ggplot(data=.,
         aes(x=stage,
             y=percent,
             fill=stage)) + 
  geom_bar(stat="identity", position = "dodge") +
  geom_text(aes(label = scales::comma(total_count)),
            position = position_dodge(width = 1),
            vjust=-0.5,
            size = 3) +
  theme_minimal() +
  scale_fill_brewer(palette="Set3") +
  ylab('Percentage') +
  xlab('Stage') +
  theme(legend.position = "none") +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1)) -> p3

df %>%
  group_by(capture_status) %>%
  summarise(total_count=n()) %>% 
  mutate(percent = prop.table(total_count)) %>% 

  ggplot(data=.,
         aes(x=reorder(capture_status, -percent),
             y=percent,
             fill=capture_status)) + 
  geom_bar(stat="identity", position = "dodge") +
  geom_text(aes(label = scales::comma(total_count)),
            position = position_dodge(width = 1),
            vjust=-0.5,
            size = 3) +
  theme_minimal() +
  scale_fill_brewer(palette="Spectral") +
  xlab('Capture status') +
  ylab('Percentage') +
  theme(legend.position = "none") +
  scale_y_continuous(labels = scales::percent_format(),
                     limits = c(0,1)) -> p4


#svg(bg = "white", width=14, height=6)
#pdf(bg = "white", width=14, height=6)
#pdf(file = "RNA_samples_plot", width = 12, height = 5)
p1 + p2 + p3 + p4
```



476,193 Hv3_splicedmasterTable.gtf
371,104 RNA_Hv3_splicedmasterTable.gtf



